architecture/2025-04-aci-confidential-containers.md

```diff
@@ -0,0 +1,388 @@
+- Full Dapr parity will not be available initially.  
+- ACI Service Discovery.
+
+### User scenarios (optional)
+
+<!--
+Describe the user scenarios for this design. Ensure that you define the
+roles and personas in these user scenarios when it requires API design.
+If you have an existing issue that describes the user scenarios, please
+link to that issue instead.
+-->
+
+The scenario document for severless is in progress [here](https://github.com/radius-project/design-notes/blob/9bcaa014c013b254593426280add5ef5c69b265e/architecture/2025-01-serverless-feature-spec.md#scenario-3-punch-through-to-platform-specific-features-and-incremental-adoption-of-radius-into-existing-serverless-applications). Please refer to the scenarios defined in that doc, specifically scenario 3 as it relates to `runtimes` punchthroughs. 
+
+## User Experience (if applicable)
+<!--
+If the change impacts the user experience, provide expected interaction 
+flow we aim to achieve through this proposal.
+
+When users interact with Radius through the CLI, include sample 
+input commands and their corresponding output. Include a bicep/helm code 
+sample, if this proposal involves updates to that experience.
+-->
+
+Ted is a developer trying to run his applications on a serverless compute platform. His application needs to run on confidential containers. His E2E experience defining a Bicep template using ACI compute looks like this: 
+
+**Setup**
+
+1. Run `rad init`.
+2. When prompted to create an environment, choose the `aci` compute option.
+3. When prompted, choose whether or not to support confidential containers. This will download the Azure CLI and the `confcom` extension. 
+
+Note: The Kubernetes flow will not change, minus the extra step to select Kubernetes compute. 
+
+**Sample Input:**
+<!--
+Provide a sample CLI command input and/or bicep/helm code.
+-->
+```
+rad init --full 
+
+Enter an environment name              
+> default  
+                                                      
+Select the compute for the environment
+  >  1. kubernetes 
+  >  2. aci
+
+Setup support for confidential containers? Selecting 'Yes' will install the Azure CLI and the `confcom` extension that are required to support confidential containers on ACI.  
+  >  1. Yes
+  >  2. No    
+```
+
+**Authoring:**
+<!--
+Provide a sample output for the inputs provided above.
+-->
+
+Ted now has a default environment setup using `ACI` as the underlying compute. He can define a Radius container that runs with a `Confidential` SKU. Ted can set additional values on the container group profile like `name`, `osType`, etc to pass through when Radius creates the container group profile. 
+
+Note: The `runtimes` property is not meant to be a replacement for defining a `ContainerGroupProfile` resource in Bicep. It is meant to be used as a punchthrough for specific values that the user wants to set on the default container group profile that Radius creates.  
```

Comment:
Awesome. So Radius will merge what's provided in the `ContainerGroupProfile` within the `runtimes` property with the default container group profile that Radius created?

When it comes to conflicts with what the user provided in `runtimes` vs. what's in the default container group profile created by Radius, I recommend that the user's values overwrite whatever the values are in the default cgp.

Reply:
nvm, i see these questions answered in the detailed design :)